com.unity.render-pipelines.high-definition/Editor/Material/Water/ShaderGraph/WaterSubTarget.cs

```diff
@@ -0,0 +1,283 @@
+                referenceName = "SHADERPASS_FORWARD",
+                lightMode = "ForwardOnlyProcedural",
+                useInPreview = true,
+
+                // Collections
+                structs = CoreStructCollections.BasicProcedural,
+                // We need motion vector version as Forward pass support transparent motion vector and we can't use ifdef for it
+                requiredFields = CoreRequiredFields.BasicLighting,
+                renderStates = WaterForward,
+                pragmas = HDShaderPasses.GeneratePragmas(CorePragmas.DotsInstanced, false, false),
+                defines = HDShaderPasses.GenerateDefines(WaterForwardDefinesProcedural, false, false),
+                includes = GenerateIncludes(),
+
+                virtualTextureFeedback = true,
+                customInterpolators = CoreCustomInterpolators.Common
+            };
+
+            IncludeCollection GenerateIncludes()
+            {
+                var includes = new IncludeCollection();
+
+                includes.Add(CoreIncludes.CorePregraph);
+                includes.Add(CoreIncludes.kNormalSurfaceGradient, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.kLighting, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.kLightLoopDef, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.kPassPlaceholder, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.kLightLoop, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.CoreUtility);
+                includes.Add(CoreIncludes.kDecalUtilities, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.kPostDecalsPlaceholder, IncludeLocation.Pregraph);
+                includes.Add(CoreIncludes.kShaderGraphFunctions, IncludeLocation.Pregraph);
+                includes.Add(WaterIncludes.kPassWaterForward, IncludeLocation.Postgraph);
+
+                return includes;
+            }
+        }
+
+        public static PassDescriptor GenerateWaterForwardPassMesh()
+        {
+            return new PassDescriptor
+            {
+                // Definition
+                displayName = "ForwardOnlyMesh",
+                referenceName = "SHADERPASS_FORWARD",
+                lightMode = "ForwardOnlyMesh",
+                useInPreview = true,
+
+                // Collections
+                structs = CoreStructCollections.Basic,
+                // We need motion vector version as Forward pass support transparent motion vector and we can't use ifdef for it
+                requiredFields = CoreRequiredFields.BasicLighting,
+                renderStates = WaterForward,
+                pragmas = HDShaderPasses.GeneratePragmas(CorePragmas.DotsInstanced, false, false),
+                defines = HDShaderPasses.GenerateDefines(WaterForwardDefinesMesh, false, false),
+                includes = GenerateIncludes(),
+
+                virtualTextureFeedback = true,
```

Comment:
I don't think water will work with procedural texturing? (otherwise it mean you need to output VT feedback texture), let's not do it for now (same in other pass)

Reply:
put false